content/chat/typing.textile

```diff
@@ -1,7 +1,91 @@
 ---
 title: Typing indicator
-meta_description: "Real time functionality to indicator when a client is typing or has stopped typing."
+meta_description: "Realtime functionality to indicate when a member is typing or has stopped typing."
 product: chat
 languages:
   - javascript
 ---
+
+Signal when a member is typing to indicate to other members that a message is being written.
+
+<img src="/images/products/chat/typing.png" alt="Example of a chat message input box, above this has two names of other members who are currently typing.">
+
+h2(#implementation). Implementation
+
+Use the "presence":/presence-occupancy/presence feature to implement a typing indicator.
+
+The member status can indicate whether they're currently typing. This status allows other members to see a visual cue, triggering an update event when a member begins typing to update the Chat UI for all members. For example, "Jane is typing...".
+
+```[javascript]
+    // Subscribe to any presence updates for the current presence set
+    channel.presence.subscribe('update', (memberUpdate) => {
+      // Event triggered for when someone has started or stopped typing.
+      // Your custom functionality is required here to show the indication someone is typing for example:
+      // - "John is typing..."
+      // - or "Multiple people are typing..."
+    });
+```
+
+Depending on the UI, it's advisable to show only a limited number of names of people who are typing, and if that number is exceeded, display a message such as "People are typing...".
+
+h3(#example). Example of typing indicators
+
+For the member, an event listener waits for a keypress before publishing an update to the presence set, updating this member's status with a string indicating they're typing. A timer is set to expire in five seconds to reset this status if no further keys have been pressed in that time.
+
+The following is an example of configuring typing indicator updates using presence:
+
+```[javascript]
+    let isTyping = false;
+    let typingInterval;
+    let usersTyping = [];
+
+    var client = new Ably.Realtime('{{API_KEY}}');
+    var channel = client.channels.get('{{RANDOM_CHANNEL_NAME}}');
+
+    // Subscribe to any presence updates for the current presence set
+    channel.presence.subscribe('update', (memberUpdate) => {
+      // find your user by their client id
+      user = '';
```

Comment:
Maybe use a non-existent function like `user = getUserByClientID(memerUpdate.clientID)` to highlight the correct way of getting the clientID or other info available in the `memberUpdate`. (I'm not sure if it's `memberUpdate.clientID` btw).

Can this be done so that it works standalone without _"find your user by their client id"_ by only using info available in presence? If names are hard maybe just using a `usersTyping` Set (not array) where we add and remove the clientIDs directly?